aya/src/util.rs

```diff
@@ -95,6 +95,36 @@ fn parse_kernel_symbols(reader: impl BufRead) -> Result<BTreeMap<u64, String>, i
     Ok(syms)
 }
 
+/// Find current system's syscall prefix by testing on the BPF syscall.
+/// If no valid value found, will return the first possible value which
+/// would probably lead to error in later API calls.
+pub fn syscall_prefix() -> Result<&'static str, io::Error>{
+    const prefixes : [&str; 7] = [
+        "sys_",
+        "__x64_sys_",
+        "__x32_compat_sys_",
+        "__ia32_compat_sys_",
+        "__arm64_sys_",
+        "__s390x_sys_",
+        "__s390_sys_",
+    ];
+    let ksym = kernel_symbols()?;
+    let values = ksym.into_values().collect::<Vec<_>>();
+    for p in prefixes {
+        if values.contains(&format!("{}bpf", p)){
+            return Ok(p);
+        }
+    }
+    return Ok(prefixes[0])
+}
+
+/// Given a name, it finds and append the system's syscall prefix to it.
+/// This function doesn't check if the name is for an existing syscall.
+/// For example, given "clone" the helper would return "sys_clone" or "__x64_sys_clone".
+pub fn syscall_fnname_add_prefix(name: &str) -> Result<String, io::Error>{
```

Comment:
I don't think we should provide this. It's essentially a more complex name for format!(), and it encourages bad perf eg:

```
let a = syscall_fnname_add_prefix(...)?;
let b = syscall_fnname_add_prefix(...)?;
let c = syscall_fnname_add_prefix(...)?;
```

Will parse and iterate over /proc/kallsyms three times, when the following doesn't:

```
let prefix = syscall_prefix()?;
let a = format!("{prefix}foo");
let b = format!("{prefix}bar");
let c = format!("{prefix}baz");
```

Reply:
Agree, the python and go implementations seem to be using a cache: https://github.com/iovisor/bcc/blob/a078cbc7cfbdcee88c9cf3e2b76c2c8c1517e478/src/cc/bcc_syms.cc#L600 
I've removed this function, it might be added later if the kernel symbols cache gets added